tests/gtfs/test_unmatched_id_warnings.py

```diff
@@ -0,0 +1,152 @@
+"""Tests for inserting invalid data into various GTFS tables."""
+import pytest
+import pandas as pd
+import numpy as np
+
+from transport_performance.gtfs.validation import GtfsInstance
+
+
+@pytest.fixture(scope="function")
+def gtfs_fixture():
+    """Test GtfsInstance() fixture."""
+    gtfs = GtfsInstance()
+    return gtfs
+
+
+@pytest.mark.runinteg
+class TestUmatchedIDWarnings(object):
+    """Tests for unmatch ID warnings in GTFS data."""
+
+    def test_unmatched_id_warnings_calendar(self, gtfs_fixture):
+        """Tests for unmatched IDs in the calendar table."""
+        # insert invalid data
+        gtfs_fixture.feed.calendar = pd.concat(
+            [
+                gtfs_fixture.feed.calendar,
+                pd.DataFrame(
+                    {
+                        "service_id": [101],
+                        "monday": [0],
+                        "tuesday": [0],
+                        "wednesday": [0],
+                        "thursday": [0],
+                        "friday": [0],
+                        "saturday": [0],
+                        "sunday": [0],
+                        "start_date": ["20200104"],
+                        "end_date": ["20230301"],
+                    }
+                ),
+            ],
+            axis=0,
+        )
+
+        expected_message = {
+            "type": "error",
+            "message": "Invalid service_id; maybe has extra space characters",
+            "table": "calendar",
+            "rows": [0],
+        }
+
+        # ensure raised errors match expected errors
+        assert expected_message == gtfs_fixture.is_valid().iloc[0].to_dict(), (
+            "GTFS validation failed to identify an unmatched ID in "
+            "the calendar table"
+        )
+
+    @pytest.mark.runinteg
+    def test_unmatched_id_warnings_trips(self, gtfs_fixture):
+        """Tests for unmatched IDs in the trips table."""
+        # insert invalid data
+        gtfs_fixture.feed.trips = pd.concat(
+            [
+                gtfs_fixture.feed.trips,
+                pd.DataFrame(
+                    {
+                        "service_id": [101],
+                        "route_id": [20304],
```

Comment:
same applies to the integer -> string formats here. Though route_id "20304" does not pre-exist and successfully raises an "undefined route_id" error, you won't see the equivalent error for service_id 101, unless you pass an id that doesn't pre-exist.